src/datagrid/DataGrid.tsx

```diff
@@ -217,6 +221,7 @@ export class DataGrid extends React.PureComponent<DataGridProps, DataGridState>
                     )}
                 </span>
                 <div className={ClassNames.DATAGRID_COLUMN_SEPARATOR} />
+
```

Comment:
DATAGRID_COLUMN_SEPARATOR is not needed in case of expandable rows. Also shift next column header text to the left so that caret icon and next header column will look like a single column.

Reply:
@vinodwani18  It is as per https://clarity.design/documentation/datagrid/expandable-rows